libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32F413xx.py

```diff
@@ -0,0 +1,858 @@
+#!/usr/bin/env python
+'''
+these tables are generated from the STM32 datasheet RM0430 in en.DM00305666.pdf for the
+STM32F413
+'''
+
+# additional build information for ChibiOS
+build = {
+    "CHIBIOS_STARTUP_MK"  : "os/common/startup/ARMCMx/compilers/GCC/mk/startup_stm32f4xx.mk",
+    "CHIBIOS_PLATFORM_MK" : "os/hal/ports/STM32/STM32F4xx/platform.mk"
+}
+
+# MCU parameters
+mcu = {
+    # location of MCU serial number
+    'UDID_START' : 0x1FFF7A10,
+
+    # ram map, as list of (address, size-kb, flags)
+    # flags of 1 means DMA-capable
+    # flags of 2 means faster memory for CPU intensive work
+    'RAM_MAP' : [
+        (0x20000000, 320, 1), # main memory, SRAM1/SRAM2 DMA safe
+
+    ],
+
+    'EXPECTED_CLOCK' : 100000000,
+
+    'DEFINES' : {
+        'STM32F4' : '1',
+    }
+}
+
+DMA_Map = {
+	# format is (DMA_TABLE, StreamNum, Channel)
+	# extracted from tabula-rm0430-stm32f413423-advanced-armbased-32bit-mcus-stmicroelectronics-dma.csv
+	"ADC1"    	:	[(2,0,0),(2,4,0)],
+	"AES_IN"  	:	[(2,6,2)],
+	"AES_OUT" 	:	[(2,5,2)],
+	"DAC1"    	:	[(1,5,7)],
+	"DAC2"    	:	[(1,6,7)],
+	"DFSDM1_FLT0"	:	[(2,6,3),(2,0,7)],
+	"DFSDM1_FLT1"	:	[(2,1,3),(2,4,3)],
+	"DFSDM2_FLT0"	:	[(2,0,8),(2,4,8)],
+	"DFSDM2_FLT1"	:	[(2,1,8),(2,5,8)],
+	"DFSDM2_FLT2"	:	[(2,2,8),(2,6,8)],
+	"DFSDM2_FLT3"	:	[(2,3,8),(2,7,8)],
+	"I2C1_RX" 	:	[(1,0,1),(1,5,1)],
+	"I2C1_TX" 	:	[(1,1,0),(1,6,1),(1,7,1)],
+	"I2C2_RX" 	:	[(1,2,7),(1,3,7)],
+	"I2C2_TX" 	:	[(1,7,7)],
+	"I2C3_RX" 	:	[(1,1,1),(1,2,3)],
+	"I2C3_TX" 	:	[(1,4,3),(1,5,6)],
+	"I2CFMP1_RX"	:	[(1,3,1),(1,0,7)],
+	"I2CFMP1_TX"	:	[(1,1,2),(1,7,4)],
+	"I2S2_EXT_RX"	:	[(1,3,3)],
+	"I2S2_EXT_TX"	:	[(1,4,2)],
+	"I2S3_EXT_RX"	:	[(1,2,2),(1,0,3)],
+	"I2S3_EXT_TX"	:	[(1,5,2)],
+	"QUADSPI" 	:	[(2,7,3)],
+	"SAI1B"   	:	[(2,4,1)],
```

Comment:
Reference manual calls this out separately to `SAI1_B` - I suspect this is a typo, I'll fix this up when I have a moment.

Reply:
I have adjusted the csv and regenerated the table.